main.py

```
@@ -6,97 +6,352 @@
     QgsVectorLayer,
     QgsGeometry,
     QgsField,
+    QgsFields,
     QgsSpatialIndex,
```

Ruff (0.8.2)

`qgis.core.QgsSpatialIndex` imported but unused

Remove unused import

(F401)